# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.h
# Commit: 78181c32bdc6
# Full Hash: 78181c32bdc6b1d66eac04f3ce11a2902f477823
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 912cf1adac80 -r 78181c32bdc6 dom/base/nsGlobalWindowInner.h
--- a/dom/base/nsGlobalWindowInner.h	Thu Feb 27 16:17:34 2025 +0000
+++ b/dom/base/nsGlobalWindowInner.h	Thu Feb 27 16:17:35 2025 +0000
@@ -130,6 +130,7 @@
 struct SizeToContentConstraints;
 class WebTaskScheduler;
 class WebTaskSchedulerMainThread;
+class WebTaskSchedulingState;
 class SpeechSynthesis;
 class Timeout;
 class TrustedTypePolicyFactory;
@@ -961,6 +962,12 @@
   bool CrossOriginIsolated() const override;
 
   mozilla::dom::WebTaskScheduler* Scheduler();
+  void SetWebTaskSchedulingState(
+      mozilla::dom::WebTaskSchedulingState* aState) override;
+  mozilla::dom::WebTaskSchedulingState* GetWebTaskSchedulingState()
+      const override {
+    return mWebTaskSchedulingState;
+  }
 
  protected:
   // Web IDL helpers
@@ -1277,6 +1284,7 @@
   RefPtr<mozilla::dom::ContentMediaController> mContentMediaController;
 
   RefPtr<mozilla::dom::WebTaskSchedulerMainThread> mWebTaskScheduler;
+  RefPtr<mozilla::dom::WebTaskSchedulingState> mWebTaskSchedulingState;
 
   RefPtr<mozilla::dom::TrustedTypePolicyFactory> mTrustedTypePolicyFactory;
 